spec/support/controller_macros.rb

```diff
@@ -0,0 +1,8 @@
+module ControllerMacros
+  def login_admin
+    before(:each) do
```

Comment:
What do you think of removing this `before(:each)` block, and calling `before(:each) { login_admin }` instead?
I think it makes it more clear what's happening.